### ISSUE TITLE
Crash when calling m.add_vehicle_type() in v.0.8.1

### ISSUE THREAD

**Issue by bmorlo** [PyVRP/PyVRP#547]
#### Describe the bug

I've just updated to v0.8.1 and now when I call  ```m.add_vehicle_type()``` in my code, the program crashes with this error:

Process finished with exit code -1073741819 (0xC0000005)

The issue does not exist when using v0.7.0.

#### Code sample

```python

import numpy as np
from scipy.spatial import distance as comp_dist
import math
import time
import random

import vrplib
from pyvrp import Model, read
from pyvrp.plotting import (
	plot_coordinates,
	plot_instance,
	plot_result,
	plot_route_schedule,
)
from pyvrp.stop import MaxIterations, MaxRuntime

from matplotlib import pyplot as plt
import seaborn as sbn
from tabulate import tabulate


def generate_all_coords(_num_poi):
	"""Returns a list of double int lists specifying all the coordinates of points of interest and the depot"""
	np.random.seed(1234)
	coords_all_nodes = []
	# Generate the locations of all points of interest + the depot
	for i in range(_num_poi + 1):
		coords_single_random_node = [np.random.randint(-1000, 1000), random.randint(-1000, 1000)]
		coords_all_nodes.append(coords_single_random_node)
	# Set the depot location to (0, 0)
	coords_all_nodes[0] = [0, 0]
	# print(coords_all_nodes)
	return coords_all_nodes


def generate_distance_matrix(_coords_all_nodes):
	"""Returns a list of distance matrix rows aka the distance matrix belonging to the nodes. Euclidean distances"""
	distance_matrix = []
	for i in range(len(_coords_all_nodes)):
		frm_x = _coords_all_nodes[i][0]
		frm_y = _coords_all_nodes[i][1]
		temp_row_of_distance_matrix = []
		for j in range(len(_coords_all_nodes)):
			to_x = _coords_all_nodes[j][0]
			to_y = _coords_all_nodes[j][1]
			temp_row_of_distance_matrix.append(math.ceil(comp_dist.euclidean([frm_x, frm_y], [to_x, to_y])))
		distance_matrix.append(temp_row_of_distance_matrix)
	# print(distance_matrix)
	return distance_matrix


def main(_num_poi, _num_vehicles, _num_planning_horizons, _runtime_max_in_seconds, _solver_names,
		 _starting_nodes_indices):
	
	coords = generate_all_coords(_num_poi)
	distances = generate_distance_matrix(coords)  # TODO: ACHTUNG WIR ARBEITEN HIER MIT EUCLIDEAN DISTANCES!!!
	
	# Create the data of the test problem
	data = {"distances": distances,
			"demands": [0] + [1] * _num_poi,
			"vehicle_capacities": [_num_planning_horizons] * _num_vehicles,
			"num_vehicles": _num_vehicles,
			"start": [0] * _num_vehicles,
			"end": [0] * _num_vehicles}
	
	"""PyVRP"""
	pyvrp_model = Model()
	pyvrp_model.add_vehicle_type(_num_vehicles, capacity=_num_planning_horizons)
	depot = pyvrp_model.add_depot(x=coords[0][0], y=coords[0][1])
	clients = [pyvrp_model.add_client(x=coords[client_idx][0], y=coords[client_idx][1],
									  delivery=data["demands"][client_idx]) for client_idx in range(1, len(coords))]
	
	locations = [depot] + clients
	
	for frm_index in range(len(locations)):
		for to_index in range(len(locations)):
			distance = distances[frm_index][to_index]
			pyvrp_model.add_edge(locations[frm_index], locations[to_index], distance=distance)
	
	result_pyvrp = pyvrp_model.solve(stop=MaxRuntime(_runtime_max_in_seconds), seed=42)
	print(f"Objective value PyVRP: {result_pyvrp.cost()}")


if __name__ == '__main__':
	"""Input the relevant CONSTANTS"""
	NUM_POINTS_OF_INTEREST = 50
	NUM_VEHICLES = 10
	NUM_PLANNING_HORIZONS = 5  # == vehicle capacity
	RUNTIME_MAX_IN_SECONDS = 30
	SOLVER_NAMES = ['OR_Routing', 'PyVRP']
	STARTING_NODES_INDICES = [1, 2, 3, 4, 5]
	main(NUM_POINTS_OF_INTEREST, NUM_VEHICLES, NUM_PLANNING_HORIZONS, RUNTIME_MAX_IN_SECONDS, SOLVER_NAMES,
		 STARTING_NODES_INDICES)

```
#### Expected Output

Program should not crash.

**Comment by leonlan** [comment]
I cannot reproduce this issue locally nor on [Google Colab](https://colab.research.google.com/drive/1zAwNpNi1hYZ5omoosbMkZs4tlwKL8iqO?usp=sharing).

Just to be sure: if you run this script on your local machine, do you get the mentioned error, or were you running a more elaborate version of this script? Also, could you run the following:

```
python -c "import pyvrp; pyvrp.show_versions()"
```

(see [here](https://pyvrp.org/setup/getting_help.html#submitting-a-bug-report) for more details).

This should return an output that looks like this:
```

INSTALLED VERSIONS
------------------
     pyvrp: 0.8.0a0
     numpy: 1.26.4
matplotlib: 3.8.3
    vrplib: 1.3.0
      tqdm: 4.66.2
     tomli: 2.0.1
    Python: 3.11.7
```

**Comment by N-Wouda** [comment]
@bmorlo to add to @leonlan's comment, which OS are you running this on?

**Comment by bmorlo** [comment]
I ran your script. I get a slightly different output. My pyvrp version shows (in my opinion) incorrectly. I updated to pyvrp 0.8.1 - even though it says pyvrp: 0.7.0. I am using Pycharm and thus always "update" my packages through it. Is this bad practice?

I am running Windows 10 Enterprise.

Thanks for your help :D

```
H:\>python -c "import pyvrp; pyvrp.show_versions()"
INSTALLED VERSIONS
------------------
     pyvrp: 0.7.0
     numpy: 2.0.0b1
matplotlib: 3.8.3
    vrplib: 1.3.0
      tqdm: 4.66.2
     tomli: 2.0.1
    Python: 3.11.8
```

**Comment by N-Wouda** [comment]
I do not know; I do not use PyCharm. I assume PyCharm manages a virtual environment for you that contains a different version of PyVRP.

I just tried to replicate this under Windows and I run into the same issue. From what I can gather it has something to do with the native extensions. I'll try to fix this and then push 0.8.2.

**Comment by N-Wouda** [comment]
I am harderning our CD to also execute the entire test suite for every wheel it builds. That now works in #548, and indeed the Windows wheels fail to execute the test suite. I'll try to figure out why.

**Comment by N-Wouda** [comment]
It's almost certainly got to do with a 32/64-bit issue that somehow pops up on Windows, but not on the other platforms. Somewhat strange because we explicitly require 64-bit integer types. I'm trying to fix this now.

**Comment by N-Wouda** [comment]
Fixed in #548. I'll merge that in a moment and then draft a release of v0.8.2!

**Comment by N-Wouda** [comment]
v0.8.2 is [currently building](https://github.com/PyVRP/PyVRP/actions/runs/8422762356) and should be on PyPI in a few minutes, if all goes well 🤞.

**Comment by N-Wouda** [comment]
Done, should be fixed now with `pyvrp==0.8.2`. I just checked on Windows, and this looks encouraging:

<img width="365" alt="image" src="https://github.com/PyVRP/PyVRP/assets/16272507/affcea2f-0327-4734-b252-0f9218d0e8e4">

I am going to yank `0.8.1` from PyPI because it is known broken. Users should not use that version anymore.

**Comment by N-Wouda** [comment]
Thanks for reporting the issue @bmorlo!
